### PR TITLE
fix(resource): fix properties sorting (DEV-1204)

### PR DESCRIPTION
**Pull request by mpro7**
resolves DEV-1204

### CODE REVIEW

File: src/app/workspace/resource/resource.component.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /* eslint-disable @typescript-eslint/member-ordering */
+import { A } from '@angular/cdk/keycodes';
```
Comment (mdelez) [inline src/app/workspace/resource/resource.component.ts:2]: can be removed, probably an auto-import

File: src/app/workspace/resource/resource.component.ts [hunk 2]
```diff
@@ -30,6 +31,7 @@ import {
     SystemPropertyDefinition
 } from '@dasch-swiss/dsp-js';
 import { Subscription } from 'rxjs';
+import { delay } from 'rxjs/operators';
```
Comment (mdelez) [inline src/app/workspace/resource/resource.component.ts:34]: can be removed, probably an auto-import

File: src/app/workspace/resource/resource.component.ts [hunk 3]
```diff
@@ -319,8 +321,11 @@ export class ResourceComponent implements OnInit, OnChanges, OnDestroy {
                             this.requestIncomingResources(this.resource);
                         }
 
-                        // gather resource property information
-                        res.resProps = this.initProps(response);
+                        setTimeout(() => {
```
Comment (mdelez) [inline src/app/workspace/resource/resource.component.ts:324]: Why does this need to be wrapped in a `setTimeout`?

**Review by mdelez — CHANGES_REQUESTED** [review]

File: src/app/workspace/resource/resource.component.ts (hunk 3 above)
Comment (mpro7) [reply src/app/workspace/resource/resource.component.ts:324]: Sorry, all of these you've find are leftovers I forgot to remove. Haven't work with TS lately and my IDE is not removing unused imports though.

**Comment by mdelez** [comment]
@mpro7 The `if` condition in the second `sort` seems to never be met so I still have the issue on my end. Is it working for you?

**Comment by mpro7** [comment]
> @mpro7 The `if` condition in the second `sort` seems to never be met so I still have the issue on my end. Is it working for you?

It's working for me. Why do you say the condition it's never to be met?

**Comment by mdelez** [comment]
> > @mpro7 The `if` condition in the second `sort` seems to never be met so I still have the issue on my end. Is it working for you?
> 
> It's working for me. Why do you say the condition it's never to be met?

The account I'm using (root) has german set as its language so the label is "hat Standoff Link zu" which is the reason why the `if` statement is never met. Also, this would mean that if anyone were to put "has" in their property label (which is pretty common unfortunately, especially for Lausanne) it would also get sorted to the bottom. 

The biggest issue I see is that these default props do not have a `guiOrder`. I took a look at a more complex resource to see what else we can sort by. I think we're safe to either sort all properties that have a `undefined` `guiOrder` to the bottom **or** we can sort by `isEditable` but I'm not certain under what conditions this boolean is set.

**Comment by mpro7** [comment]
> > > @mpro7 The `if` condition in the second `sort` seems to never be met so I still have the issue on my end. Is it working for you?
> > 
> > 
> > It's working for me. Why do you say the condition it's never to be met?
> 
> The account I'm using (root) has german set as its language so the label is "hat Standoff Link zu" which is the reason why the `if` statement is never met. Also, this would mean that if anyone were to put "has" in their property label (which is pretty common unfortunately, especially for Lausanne) it would also get sorted to the bottom.
> 
> The biggest issue I see is that these default props do not have a `guiOrder`. I took a look at a more complex resource to see what else we can sort by. I think we're safe to either sort all properties that have a `undefined` `guiOrder` to the bottom **or** we can sort by `isEditable` but I'm not certain under what conditions this boolean is set.

Hmm, the language factor I haven't took into the consideration. That's why the property labels should be unified and in English only ;)

Anyways, I will try to change the condition and see how if it works with `undefined`. It seems to be the most reasonable.

**Review by mdelez — APPROVED** [review]
works well, thanks :)

**Pull request merged**